demos/demo-react-ts/src/components/screens/CallingScreen.tsx

```diff
@@ -38,6 +39,7 @@ function CallingScreen({
   const [toggleKeypad, setToggleKeypad] = useState(false);
 
   const handleToggleRecord = () => {
+    setIsCallRecorded(true);
```

Comment:
Seems like this should either be a toggle, or the button should become disabled once you've clicked it
```suggestion
    setIsCallRecorded(!isCallRecorded);
```

Reply:
this setIsCallRecorded coming from App.tsx. The use for this to tell App that user clicked start recording button which means we need to send recording url in engagement. the button functionality is handled separately by different toggle.

Reply:
I believe `handleToggleRecord` is invoked when the toggle button is clicked. Right now, we are sending `hs_call_recording_url` upon the user's first click. What should happen when the user clicks twice?

Reply:
This is just to track that call record button was clicked. it doesnt matter how many times it is click. hence, it gets set to true in the state. It gets reset once call is completed/saved.